### PR TITLE
build_refacto

### PR DESCRIPTION
**Pull request by erebe**

### CODE REVIEW

File: src/cmd/docker.rs [hunk 1]
```diff
@@ -0,0 +1,623 @@
+use crate::cmd::command::{CommandError, QoveryCommand};
+use crate::errors::EngineError;
+use crate::events::EventDetails;
+use chrono::Duration;
+use std::path::Path;
+use std::process::ExitStatus;
+use url::Url;
+
+#[derive(thiserror::Error, Debug)]
```
Comment (benjaminch) [inline src/cmd/docker.rs:9]: Nice ! :)

File: src/build_platform/local_docker.rs [hunk 2]
```diff
@@ -133,27 +115,25 @@ impl LocalDocker {
...
-        };
-
-        docker_args.push(into_dir_docker_style);
-
-        // docker build
-        let mut cmd = QoveryCommand::new("docker", &docker_args, &self.get_docker_host_envs());
-
-        let exit_status = cmd.exec_with_abort(
-            Duration::minutes(BUILD_DURATION_TIMEOUT_MIN),
+        // FIXME: pass a Vec<(key, value)> instead of spliting always the string
```
Comment (deimosfr) [inline src/build_platform/local_docker.rs:118]: Did you forget to fix?

File: src/build_platform/local_docker.rs [hunk 3]
```diff
@@ -414,67 +386,10 @@ impl BuildPlatform for LocalDocker {
...
-        );
-
-        Ok(result)
-    }
-
     fn build(
         &self,
         build: Build,
-        force_build: bool,
+        _force_build: bool,
```
Comment (deimosfr) [inline src/build_platform/local_docker.rs:392]: please remove this if not used anymore

File: src/cmd/docker.rs [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+    }
+
+    #[test]
+    fn test_docker_build() {
+        // start a local registry to run this test
+        // docker run --rm -d -p 5000:5000 --name registry registry:2
+        let docker = Docker::new_with_options(false, None).unwrap();
+        let image_to_build = ContainerImage {
+            registry: private_registry_url(),
+            name: "erebe/alpine".to_string(),
```
Comment (deimosfr) [inline src/cmd/docker.rs:538]: please use qovery repo instead

File: src/cmd/docker.rs [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+            tags: vec!["3.15".to_string()],
+        };
+        let image_cache = ContainerImage {
+            registry: private_registry_url(),
+            name: "erebe/alpine".to_string(),
```
Comment (deimosfr) [inline src/cmd/docker.rs:543]: same

File: src/errors/mod.rs [hunk 6]
```diff
@@ -2286,6 +2287,24 @@ impl EngineError {
...
+    /// Creates new error from an Docker error
+    ///
+    /// Arguments:
+    ///
+    /// * `event_details`: Error linked event details.
+    /// * `error`: Raw error message.
+    pub fn new_docker_error(event_details: EventDetails, error: DockerError) -> EngineError {
+        EngineError::new(
+            event_details,
+            Tag::HelmChartUninstallError,
```
Comment (deimosfr) [inline src/errors/mod.rs:2299]: are you sure about this tag ?

File: src/build_platform/local_docker.rs (hunk 2 above)
Comment (erebe) [reply src/build_platform/local_docker.rs:118]: I haven't finished yet :P wanted to check the test first
Going to do it

File: src/cmd/docker.rs (hunk 4 above)
Comment (erebe) [reply src/cmd/docker.rs:526]: it is on local registry, not dockerhub

File: src/errors/mod.rs (hunk 6 above)
Comment (erebe) [reply src/errors/mod.rs:2299]: goot catch :+1:

**Comment by erebe** [comment]
seen with enzo, not releasing, but to unstuck PR stacks

**Pull request merged**